File: CCC 2003/Junior/J2 - PicturePerfect.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        ArrayList<String> messages = new ArrayList<>();
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int length,perimeter, width, area = 1;
        boolean valid = true;
        while(valid) {
            area = Integer.parseInt(input.readLine());
            if(area == 0){
                break;
            }
            length = (int) Math.sqrt(area);
            width = Math.floorDiv(area, length);
            perimeter = 2*length + 2*width;
            //System.out.println("Minimum perimeter is " + 2 * length + 2 * width + " with dimensions " + length + " x " + width);
            messages.add("Minimum perimeter is " + perimeter + " with dimensions " + length + " x " + width);
            length = 0;
            width = 0;
        }
        for (String i : messages) {
            System.out.println(i);
        }
    }
}

File: CCC 2003/Junior/J2 - PicturePerfect.py
import sys
import math
input = sys.stdin.readline

n = 1
areas = []
while n != 0:
    n = int(input())
    areas.append(n)

for area in areas[:-1]:
    dividents = []
    for i in range(1, int(math.sqrt(area)+1)):
        if area % i == 0:
            dividents.append(i)
            dividents.append(area/i)
    dividents.sort()

    length = int(dividents[int(len(dividents)/2)])
    width = int(area // length)
    perimeter = 2*length + 2*width
    
    print('Minimum perimeter is {} with dimensions {} x {}'.format(perimeter, length, width))
